src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -236,6 +241,19 @@ export const useSearch = (services: DiscoverViewServices) => {
           elapsedMs,
         },
       });
+
+      // Add tracking for datasource type and query if the query is not same as last query
+      const query = searchSource.getField('query');
+      const queryStringHistory = data.query.queryString.getQueryHistory();
+      if (
+        query &&
+        !isEqual(query, queryStringHistory.slice(-1)[0]) &&
+        query.dataset?.type &&
+        query?.language
+      ) {
+        trackUiMetric(`${query?.dataset?.type}_${DATASOURCE_METRIC_SUFFIX}`);
+        trackUiMetric(`${query?.language}_${LANGUAGE_METRIC_SUFFIX}`);
```

Comment:
i never got the chance to add it for sql and ppl but here is how we were doing it for DQL

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/ui/query_editor/query_editor.tsx#L202

Reply:
For DQL they were using a custom version of usageCollector and this Api route basically carters to it. For our purposes the default Ui Metric usage collector is good enough. Therefore we are using the `reportUiStats` methods that does server side batching and writes the report onto the savedObject every 60s.